### DISCUSSION TITLE
Converting UnsafePointers to string

### DISCUSSION THREAD

**Discussion by targetlucked69** [denoland/deno#14248] [Q&A]
Hello, playing round the unstable Deno FFI and so far loving it. One question tho

Suppose I have this 

```ts
const sys = Deno.dlopen(lib, {
  func: {
    name: 'some_func',
    parameters: ['pointer'], 
    result: 'pointer' 
  },
});

sys.symbols.func() // returns UnsafePointer { value: 105553155269408n }
```

Is there a method we can use to convert the pointer to a string for example? Thanks!

**Comment by melkir** [comment]
Hello @walmartwarlord,
You might want to take a look at these resources, especially the `getCString` method.
https://deno.com/blog/v1.17#updates-to-the-ffi-api
https://doc.deno.land/deno/unstable/~/Deno.UnsafePointer
https://doc.deno.land/deno/unstable/~/Deno.UnsafePointerView

From what I understand, you have to create a buffer to store the value. On top of that you add a pointer, and the view part allows you to retrieve the value.

Edit: You can also use rust code to write your function + [deno_bindgen](https://github.com/denoland/deno_bindgen) to automatically generate the Typescript equivalent. It's a slightly different approach, but it helps to understand how it works under the hood.

**Reply by targetlucked69** [reply]
Thank you for the resource.

I've been looking at how [deno_bindgen](https://github.com/denoland/deno_bindgen) returns structs and arrays and based on the examples it looks like Rust returns stringified objects/arrays so that Deno can just JSON.parse it?
